quinn-proto/src/connection/mod.rs

```diff
@@ -3437,6 +3434,17 @@ impl Connection {
             || !self.datagrams.outgoing.is_empty()
     }
 
+    /// Update counters to account for a packet becoming acknowledged, lost, or abandoned
+    fn remove_in_flight(&mut self, pn: u64, packet: &SentPacket) {
+        // Visit known paths from newest to oldest to find the one `pn` was sent on
+        for path in [&mut self.path].into_iter().chain(self.prev_path.as_mut()) {
+            if path.first_packet.map_or(false, |first| first <= pn) {
+                path.in_flight.remove(packet);
+                return;
+            }
```

Comment:
Should this also be a method on `PathData`?

Feels like `first_packet` could/should remain private.

Reply:
Yep, that works out nicely and also lets us privatize more of `InFlight`.